src/main/scala/org/ergoplatform/nodeView/wallet/persistence/WalletStorage.scala

```diff
@@ -1,53 +1,36 @@
 package org.ergoplatform.nodeView.wallet.persistence
 
-import java.io.File
-
-import com.google.common.primitives.Ints
+import com.google.common.primitives.{Ints, Shorts}
 import org.ergoplatform.nodeView.state.{ErgoStateContext, ErgoStateContextSerializer}
+import org.ergoplatform.nodeView.wallet.scanning.{ExternalAppRequest, ExternalApplication, ExternalApplicationSerializer}
 import org.ergoplatform.settings.{Constants, ErgoSettings}
-import org.ergoplatform.wallet.secrets.{DerivationPath, DerivationPathSerializer}
-import org.ergoplatform.{ErgoAddress, ErgoAddressEncoder, P2PKAddress}
-import org.iq80.leveldb.Options
-import scorex.db.LDBFactory.factory
+import org.ergoplatform.wallet.secrets.{DerivationPath, DerivationPathSerializer, ExtendedPublicKey, ExtendedPublicKeySerializer}
+import org.ergoplatform.{ErgoAddressEncoder, P2PKAddress}
 import scorex.crypto.authds.ADDigest
-import scorex.crypto.hash.{Blake2b256, Digest32}
-import scorex.db.LDBKVStore
+import scorex.crypto.hash.Blake2b256
+import org.ergoplatform.wallet.Constants.{ApplicationId, PaymentsAppId}
+import scorex.db.{LDBFactory, LDBKVStore}
 
-import scala.util.Success
+import scala.util.{Success, Try}
 
 /**
-  * Persists version-agnostic wallet actor's mutable state.
+  * Persists version-agnostic wallet actor's mutable state
+  *  (so data which do not have different versions unlike blockchain-related objects):
+  *
+  * * tracked addresses
+  * * derivation paths
+  * * changed addresses
+  * * ErgoStateContext (is it version-agnostic?)
```

Comment:
Please define what "version agnostic" means. The question in ScalaDoc looks suspicious.

Reply:
comments improved, ErgoStateContext updates checked